### PR TITLE
New Speial Issues on Legacy Appeals

### LINKED ISSUE #13396: Legacy Appeals: Add special issues
## User or job story
User story: As a attorney and or/judge, I need to be able to flag the following special issues, so that the data can be captured for reporting purposes: 
- Burn Pit
- Military Sexual Trauma (MST)
- Blue Water
- US Court of Appeals for Veterans Claims (CAVC)

A "No Special Issues" checkbox needs to be added as a selection must be made on this page.

## Acceptance criteria
- [x] Please put this work behind the feature toggle: `:special_issues_revamp`
- [x] This feature should be accessible to the user groups with current access to Special Issues: Judges, Attorneys, & BVA Dispatch
- [x] Attorneys are able to select the above new special issues that apply to a case during the legacy workflow before submitting the case to a judge
- [x] Attorneys are able to select the "No Special Issues" checkbox if no issues apply to the case during the legacy workflow before submitting the case to a judge
- [x] Judges are able to edit/add special issues during the legacy workflow that attorneys identify apply to a case before submitting the case to dispatch
- [x] Include screenshot(s) in the Github issue if there are front-end changes
- [ ] Update documentation: Need to update training documentation
- [x] Special issues data is acceptable for reporting team

### Designs

**UPDATE The below mockup is outdated. Please move the "No Special Issues" checkbox above "About the Appellant"**

Legacy Appeals_Special Issues

### Background/context
Under law and regulations, a Veteran who served in Vietnam from January 9, 1962, to May 7, 1975 is presumed to have been exposed to herbicides.  Furthermore, a particular set of medical conditions are presumed to be service connected for anyone who has been exposed to herbicide.  Historically, the definition of "served in Vietnam" has been limited to Veterans who served on the landmass of Vietnam or its inland waterways, also know as brown water.  

Last year both a court decision and bill passed by Congress expanded the presumption of exposure to herbicide during that time period to any Veteran who served within not more than 12 nautical miles from the coast of Vietnam, also know as blue water.  

The goal of this project is to provide the Board for the Board of Veterans’ Appeals the ability to flag/mark Blue Water along with MST, Burn PIT, and CAVC contentions in Caseflow for data capture purposes only. 

### Technical notes
This data is needed for reporting purposes in Tableau so both engineering teams will need to synch to ensure the data we are providing is in the format they are requiring. 

Tracking specialty appeals like this was common practice in VACOLS for legacy appeals, see image below.  The Board is likely looking for something similar that would be completed by an attorney at the time they close their writing task and send their appeal to their judge for review.

### Resources/other links
[Product Brief ](https://docs.google.com/document/d/15izpJwUTYP7yE0SHrdLP1tsGmbLGdIOqpEOHDyLwqDo/edit#)

### PR DESCRIPTION
**Pull request by lomky**
Resolves #13396 

First of the PR stack, see later PRs for commentary on those sections.
This PR: Backend addition of new special issues
#13874: New Special Issues, Enabler middleware
#13876: User New SpecialIssues in Filters & Queue
#13877: New SpecialIssues in Dispatch EstablishClaim

### Description
 - New special issues added to the database
 - New special issues allowed in the controller

### Acceptance Criteria
- [ ] Legacy Appeals can be associated with the new special Issues

### Testing Plan
- [ ] Tests pass
 - [ ] add new special issues via the rails console
  - `legacy_appeal.special_issue_list = SpecialIssueList.create({blue_water: true})`
  - `legacy_appeal.special_issues_list`

### User Facing Changes
None
 
### Database Changes
*Only for Schema Changes*

* [x] Column comments updated
* [x] Query profiling performed (eyeball Rails log, check bullet and fasterer output)
* [x] Appropriate indexes added (especially for foreign keys, polymorphic columns, and unique constraints)
* [x] DB schema docs updated with `make docs`
* [x] #appeals-schema notified with summary and link to this PR

### Description
This PR Stack adds:
 - 4 new Special Issues
 - The new **No Special Issues** option
 - locks these changes to the `special_issues_revamp` FeatureToggle
 - Access to these Special Issues via Queue
 - Access to these Special Issues via Dispatch

### Acceptance Criteria
- With the FeatureToggle off
  - [x] No visible changes in Queue
  - [ ] No visible changes in Dispatch
- With the FeatureToggle on
  - [ ] In Queue Attorney Checkout, 4 new options & **No Special Issues** are available
  - [ ] In Queue Judge Checkout, 
    - [ ] 4 new options & **No Special Issues** are available
    - [ ] Selections made by the Attorney are preserved
  - [ ] In Dispatch Establish Case
    - [ ] 4 new options & **No Special Issues** are available
    - [ ] Selections made by the Attorney & Judge are preserved

#### FeatureToggle Off
Set the FeatureToggle to disabled in the rails console
```
> FeatureToggle.disable!(:special_issues_revamp)
```
- Become attorney `Steve Casper`
  + Load `/queue` and select a legacy case (reseed if needed)
    * Select `Decision Ready for Review`
      - [x] Confirm new special issues do not show
- Become VLJ `Abshire`
  + Load `/queue` and select a legacy case for review (reseed if needed)
    * Select `Ready for Dispatch`
      - [x] Confirm new special issues do not show
- Become `Dispatch User` `White`
  + Load `/dispatch` and hit `Establish New Claim` (reseed if needed)
    - [x] Scroll down, confirm new special issues do not show

#### FeatureToggle On
Set the FeatureToggle to disabled in the rails console
```
> FeatureToggle.enable!(:special_issues_revamp)
```
- Become attorney `Steve Casper`
  + Load `/queue` and select a legacy case (reseed if needed)
    * Select `Decision Ready for Review`
      - [x] Confirm new special issues show
      - [x] Confirm you cannot move on without selecting at least one issue
      - [x] Select an issue or two
      - [x] Finish checkout, noting the VLJ
- Become the VLJ you returned the case to
  + Load `/queue` and select the same legacy case for review
    * Select `Ready for Dispatch`
      - [x] Confirm new special issues show
      - [x] Confirm the selections the attorney made are still checked
      - [x] Uncheck all issues, confirm you cannot move on without selecting at least one issue
      - [x] Select an issue or two
      - [x] Finish checkout without errors
      - (I do not yet know how to pull a specific case in dispatch, so confirm in the database that your case's special issues are saved to the database)
        - `LegacyAppeal.find_by(vacols_id: vacols_id).special_issue_list` shows the select issues as true
- Become `Dispatch User` `White`
  + Load `/dispatch` and hit `Establish New Claim` (reseed if needed)
    - [x] Scroll down, confirm new special issuesshow
    - [x] Complete dispatch successfully

### User Facing Changes

 - With FeatureToggle disabled, **no visual changes**
 - With FeatureToggle enabled:

User & Page | Before | After
------------------|----------|--------
Attorney @ Checkout | Screenshot_2020-04-03 Draft Decision Select special issues(1)  | Screenshot_2020-04-03 Draft Decision Select special issues
Judge @ Checkout | Screenshot_2020-04-03 Draft Decision Select special issues(3)  | Screenshot_2020-04-03 Draft Decision Select special issues(2)  
Dispatch User @ Establish Claim | Screenshot_2020-04-03 Caseflow Dispatch(1)  | Screenshot_2020-04-03 Caseflow Dispatch
 
### Documentation Updates
- [x] Topline File Descriptions Added or Updated as Needed

### Database Changes
*Only for Schema Changes*

* [x] Column comments updated
* [x] Query profiling performed (eyeball Rails log, check bullet and fasterer output)
* [x] Appropriate indexes added (especially for foreign keys, polymorphic columns, and unique constraints)
* [x] DB schema docs updated with `make docs`
* [x] #appeals-schema notified with summary and link to this PR

### CODE REVIEW

File: app/controllers/special_issues_controller.rb [hunk 1]
```diff
@@ -48,7 +48,8 @@ def special_issue_params
               :foreign_pension_dic_all_other_foreign_countries,
               :foreign_pension_dic_mexico_central_and_south_america_caribb,
               :us_territory_claim_american_samoa_guam_northern_mariana_isla,
-              :us_territory_claim_puerto_rico_and_virgin_islands)
+              :us_territory_claim_puerto_rico_and_virgin_islands,
+              :burn_pit, :military_sexual_trauma, :blue_water, :cavc, :no_special_issues)
```
[the remark this answers is not in the document]
Comment (lomky) [reply app/controllers/special_issues_controller.rb:52]: https://github.com/department-of-veterans-affairs/caseflow/issues/13396#issuecomment-608607589

File: app/views/queue/index.html.erb [hunk 2]
```diff
@@ -16,7 +16,7 @@
     hasCaseDetailsRole: current_user.roles.include?('Case Details'),
     userCanViewHearingSchedule: current_user.can_view_hearing_schedule?,
     featureToggles: {
-        special_issues_revamp: FeatureToggle.enabled?(:special_issues_revamp, user: current_user)
+      special_issues_revamp: FeatureToggle.enabled?(:special_issues_revamp, user: current_user)
```
Comment (yoomlam) [inline app/views/queue/index.html.erb:19]: `special_issues_revamp` inconsistent with `specialIssuesRevamp` in `show.html.erb`.

File: docs/schema/caseflow.csv [hunk 3]
```diff
@@ -882,8 +885,10 @@ special_issue_lists,id,integer (8) PK,x,x,,,,
 special_issue_lists,incarcerated_veterans,boolean,,,,,,
 special_issue_lists,insurance,boolean,,,,,,
 special_issue_lists,manlincon_compliance,boolean,,,,,,
+special_issue_lists,military_sexual_trama,boolean,,,,,,Military Sexual Trauma (MST)
```
Comment (yoomlam) [inline docs/schema/caseflow.csv:888]: `military_sexual_trama` typo

File: db/schema.rb [hunk 4]
```diff
@@ -1134,8 +1137,10 @@
     t.boolean "incarcerated_veterans", default: false
     t.boolean "insurance", default: false
     t.boolean "manlincon_compliance", default: false
+    t.boolean "military_sexual_trauma", default: false, comment: "Military Sexual Trauma (MST)"
     t.boolean "mustard_gas", default: false
     t.boolean "national_cemetery_administration", default: false
+    t.boolean "no_special_issues", default: false, comment: "Affirmative no special issues, added belatedly"
```
Comment (yoomlam) [inline db/schema.rb:1143]: Opinion: I think it's sufficient to say for the `comment: "No special issue for the appeal"`.

**Pull request merged**